src/transform_output/transform_output.cpp

```diff
@@ -0,0 +1,219 @@
+/** \brief Output Transformation source file
+ *
+ *  Authors      : Syed Omar Azam, Ammar Al-Mahbashi, Mohammed Ali
+ *  Date Created : 15-Aug-2019
+ *  Version      : 1.0
+ *
+ *  This function is used to transform the ABP simulator generated output 
+ *  to a table structured and readable format. It takes as input the 
+ *  file you want to transform and the name of the output file and generates 
+ *  the new output file.
+ *  
+ *  This function can generate formatted output for the following components
+ *      1- ABP Simulator
+ *      2- Sender
+ *      3- Receiver
+ *      4- Subnet
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <iostream>
+#include <fstream>
+#include <iomanip>
+
+#include "../../include/transform_output/transform_output.hpp"
+
+using namespace std;
+
+/**
+ * This function is used to trim spaces present on the right side of a string.
+ * It takes two parameters as below. For each line read from ip_file_name
+ * it removes the spaces on the right side so we generate readable output.
+ * @param str of type string
+ * @param chars of type string
+ */
+std::string& rtrim(std::string& str, const std::string& chars = "\t\n\v\f\r ")
+{
+    str.erase(str.find_last_not_of(chars) + 1);
+    return str;
+}
+
+void transform_output(const char *ip_file_name, const char *op_file_name) {
+	
+	/**

```

Comment:
If you are considering this for the document then its ok 
Else you can use /* coment*/ for commenting variables

Reply:
fixed